google-cloud-firestore/samples/query_watch.rb

```diff
@@ -0,0 +1,174 @@
+# Copyright 2020 Google, Inc
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require "google/cloud/firestore"
+
+def listen_document project_id:, collection_path: "cities"
+  # project_id = "Your Google Cloud Project ID"
+  # collection_path = "cities"
+
+  firestore = Google::Cloud::Firestore.new project_id: project_id
+  # [START fs_listen_document]
+
+  # Create an Event for notifying main thread.
+  require "concurrent"
+  event = Concurrent::Event.new
+
+  doc_ref = firestore.col(collection_path).doc "SF"
+
+  # Watch the document.
+  listener = doc_ref.listen do |snapshot|
+    puts "Received document snapshot: #{snapshot.document_id}"
+    event.set
```

Comment:
While this makes sense in tests, this pattern (listen and wait with a timeout for the first and only event) is not really something we should advertise (as we have `get()` methods for this). Instead, an action should be taken for every snapshot.

Since this does showcase how the API can be used, I don't feel strongly here. I just wanted to raise this as code like this should not exist in a production app.

Reply:
I will change this back to the callback updating a collection that is checked by the main thread, a pattern used already [in the acceptance tests](https://github.com/googleapis/google-cloud-ruby/blob/master/google-cloud-firestore/acceptance/firestore/watch_test.rb#L26-L28). Please note that I adapted the `event` from the [Python sample in master](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/firestore/cloud-client/snippets.py#L687), in case you want to raise the issue there as well.

Reply:
Done.